### DISCUSSION TITLE
How to prevent old packages from being deleted if the new port fails to build?

### DISCUSSION THREAD

**Discussion by asomers** [freebsd/poudriere#1095] [Q&A]
With `ATOMIC_PACKAGES_REPOSITORY=yes` and `COMMIT_PACKAGES_ON_FAILURE=yes`, then if a particular port gets broken by an update, `poudriere bulk` will delete the old package even though it failed to build a new one.  OTOH, with `COMMIT_PACKAGES_ON_FAILURES=no`, then a single broken port will prevent `poudriere bulk` from publishing _any_ packages, even ones build successfully.

Is there any way to ensure that `poudriere bulk` will publish all successfully-built packages, but not delete packages that failed to build?  I can't find a way.

**Comment by bdrewery** [comment]
There is `KEEP_OLD_PACKAGES` that might help. We could change the build to not delete the package until the build succeeds. However its dependencies may have been upgraded and the package, that fails to build and we keep, no longer would work.

**Comment by asomers** [comment]
I saw `KEEP_OLD_PACKAGES`.  But from the description, it sounds like the old packages won't be available automatically.  The operator would have to change the entire repository to get the old packages back, and it wouldn't be possible to mix old packages with new ones.

> its dependencies may have been upgraded and the package, that fails to build and we keep, no longer would work.

Does that mean that what I'm asking for is just fundamentally a bad idea, and I should abandon it?
